### PR TITLE
Add a new client / server command to rename CIDR

### PR DESCRIPTION
**Pull request by skywhale**
Closes https://github.com/tonarino/innernet/issues/309

### CODE REVIEW

File: server/src/db/cidr.rs [hunk 1]
```diff
@@ -106,6 +112,22 @@ impl DatabaseCidr {
         Ok(Cidr { id, contents })
     }
 
+    /// Update self with new contents, validating them and updating the backend in the process.
+    pub fn update(&mut self, conn: &Connection, contents: CidrContents) -> Result<(), ServerError> {
```
Comment (strohel) [inline server/src/db/cidr.rs:116]:
```suggestion
    /// Update self with new contents, validating them and updating the backend in the process.
    /// Currently this only supports updating the name and ignores changes to any other field.
    pub fn update(&mut self, conn: &Connection, contents: CidrContents) -> Result<(), ServerError> {
```

File: client/src/main.rs [hunk 2]
```diff
@@ -724,6 +737,37 @@ fn add_cidr(interface: &InterfaceName, opts: &Opts, sub_opts: AddCidrOpts) -> Re
...
+
+    if let Some((cidr_request, old_name)) = prompts::rename_cidr(&cidrs, &sub_opts)? {
+        log::info!("Renaming CIDR...");
+
+        let id = cidrs
+            .iter()
+            .filter(|c| c.name == old_name)
+            .map(|c| c.id)
+            .next()
+            .ok_or_else(|| anyhow!("CIDR not found."))?;
```
Comment (strohel) [inline client/src/main.rs:760]: I've tested this compiles:
```suggestion
        let id = cidrs
            .iter()
            .find(|c| c.name == old_name)
            .ok_or_else(|| anyhow!("CIDR not found."))?
            .id;
```

**Review by strohel — APPROVED** [review]
Oh great! Adding just 2 little technicalities, but approving already.

I was about to ask about making sure that CIDR names are unique, but I see the database takes care of it (same as in peer names), which is good, as that's probably the only non-racy way to check it.

File: server/src/main.rs [hunk 3]
```diff
@@ -460,6 +470,27 @@ fn add_cidr(
...
+    let cidrs = DatabaseCidr::list(&conn)?;
+
+    if let Some((cidr_request, old_name)) = shared::prompts::rename_cidr(&cidrs, &opts)? {
+        let db_cidr = DatabaseCidr::list(&conn)?
+            .into_iter()
+            .find(|c| c.name == old_name)
+            .ok_or_else(|| anyhow!("CIDR not found."))?;
+        db::DatabaseCidr::from(db_cidr).update(&conn, cidr_request)?;
+    } else {
+        println!("exited without creating CIDR.");
```
Comment (mcginty) [inline server/src/main.rs:488]:
```suggestion
        println!("exited without renaming CIDR.");
```

**Review by mcginty — COMMENTED** [review]
Looks good! One small copy-pasta typo fix.

**Comment by skywhale** [comment]
@strohel @mcginty thank you for your review :)

I commited the suggested changes and checked one more time that docker tests succeed for the new test case.

**Review by strohel — APPROVED** [review]
LGTM++

**Review by mcginty — APPROVED** [review]
🙃

**Pull request merged**